Controllers/UserProfileController.cs

```diff
@@ -47,6 +48,30 @@ public IActionResult GetWithRoles()
         }).OrderBy(up => up.UserName));
     }
 
+    [HttpGet("withroles/{id}")]
+    [Authorize(Roles = "Admin")]
+    public IActionResult GetWithRolesById(int id)
+    {
+        return Ok(_dbContext.UserProfiles
```

Comment:
Separate the logic for getting the User from the return statement, so that if no User with that Id is found, `NotFound()` can be returned